lib/validation.js

```diff
@@ -22,12 +22,27 @@ function compileSchemasForSerialization (context, compile) {
         throw new Error('response schemas should be nested under a valid status code, e.g { 2xx: { type: "object" } }')
       }
 
-      acc[statusCode] = compile({
-        schema,
-        url,
-        method,
-        httpStatus: statusCode
-      })
+      if (schema.content && !schema.content.type) {
```

Comment:
I wanted to make sure that the content is meant to be used for the purpose of having different content types. 

without this check, the schema below would break the code: 
```JS
schema: {
  response: {
    201: {
      content: { type: 'string' }
    },
  }
}
```

Is there any better way to ensure this other than `!schema.content.type`?

Reply:
I think you would need to actually check if the content is of shape media type. Or else I guess also following will break:

```js
schema: {
  response: {
    201: {
      content: { const: 'bla' }
    },
  }
}
```

So you could imho only ducktype it by checking if the attribute schema is set.

```js
if (schema.content) {
  const contentTypes = Object.keys(schema.content)
  const firstEntry = schema.content[contentTypes[0]]
  if (typeof firstEntry === 'object' && firstEntry.schema) {
    // content media type
  } else {
  // not a content media type
  }
}
```

Reply:
I would rather deprecate the shorten form. But it require team consensus.

Reply:
I also feel that deprecating the shortened form could be better, as it's already sometimes limited and added some complexities in the code, such schema is not allowed for example. 

```JS
schema: {
  response: {
    201: {
       properties: { name: String }
    },
  }
}
```

Anyway, if the team decided not to deprecate it, I will ducktype it as @Uzlopak explained

Reply:
Well. Deprecating means to still support the old style and not ripping it out(breaking change)

So we have to support the "old" style. 

But we can decide to deprecate it now and remove it in v5 in 2023.

Reply:
Btw. The snippet i provided is of course just to give an idea how to implement it. I assume it can be done more elegant. ;)

Reply:
I'm -1 in deprecating the shorter form.
I can be ok in assuming it's always JSON.

Reply:
So changing the ducktyping without deprecation of the shorter form has to be done.

Reply:
yes